src/Algebra/Properties/Semiring/Exp/TailRecursiveOptimised.agda

```diff
@@ -0,0 +1,82 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Exponentiation over a semiring optimised for tail-recursion.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra
+open import Data.Nat.Base as ℕ using (zero; suc)
+import Data.Nat.Properties as ℕ
+open import Relation.Binary
+
+module Algebra.Properties.Semiring.Exp.TailRecursiveOptimised
+  {a ℓ} (S : Semiring a ℓ) where
+
+open Semiring S renaming (zero to *-zero)
+open import Relation.Binary.Reasoning.Setoid setoid
+open import Algebra.Properties.Semiring.Exp S as U
+  using (_^_)
+
+------------------------------------------------------------------------
+-- Re-export definition from the monoid
+
+open import Algebra.Definitions.RawSemiring rawSemiring public
+  using (_^[_]*_; _^ᵗ_)
```

Comment:
It occurs to me that it would be better to simply rename the operator to `_^ᵗ_` here (as well as the accompanying properties)? It's pretty unlikely users will want to import multiple different versions of exponentiation so let's make their life easier for them. In the unlikely case they do want to import multiple versions they can always qualify them.

Reply:
Do you mean rename to `_^_`?

Reply:
Yes, sorry `_^ᵗ_` to `_^_`